app/lib/actions/recurring_action.rb

```diff
@@ -0,0 +1,62 @@
+module Actions
+  class RecurringAction < Actions::ActionWithSubPlans
+
+    include ::Dynflow::Action::Cancellable
+
+    middleware.use Actions::Middleware::KeepCurrentUser
+    middleware.use Actions::Middleware::InheritTaskGroups
+
+    def delay(delay_options, recurring_logic, action, *args)
+      task.add_missing_task_groups(recurring_logic.task_group)
+      action_class = action.is_a?(Class) ? action : action.constantize
+      action = action.to_s if action.is_a?(Class)
+      action_class.serializer_class.new [recurring_logic, action] + args
+    end
+
+    def plan(recurring_logic, action, *args)
+      serialized_args = serializer_class.new(args).tap { |s| s.perform_serialization! }.serialized_args
+      if recurring_logic.new_record?
+        plan_action(RecurringLogic::Create, recurring_logic)
```

Comment:
This seems like something we could do in a model (to initiate the delayed execution), to limit the behaviour of this action to only run the sub-task and delay the next execution.